packages/core/src/app/payment/paymentMethod/PaymentMethodV2.spec.tsx

```diff
@@ -101,4 +101,28 @@ describe('PaymentMethod', () => {
 
         expect(componentFallback.find(Foo)).toBeDefined();
     });
+
+    it('returns payment method v1 if gateway is mollie', () => {
+        const Foo: FunctionComponent<PaymentMethodProps> = ({ method }) => <div>{method.id}</div>;
+
+        const resolver = (method: PaymentMethod) => {
+            return method.id === 'applepay' ? Foo : undefined;
+        };
+
+        const componentFallback = mount(
+            <ContextProvider>
+                <PaymentMethodComponent
+                    method={{
+                        ...getPaymentMethod(),
+                        gateway: 'mollie',
+                        id: 'test',
```

Comment:
Should the id be `applepay` just to keep the test aligned with the requirement?

Reply:
No, because if I changed it to `applepay` then the resolver: 
```
 const ResolvedPaymentMethod = resolveComponent({
        id: method.id,
        gateway: method.gateway,
        type: method.type,
    });

will get applepay payment method (from V2), and the validation of: https://github.com/bigcommerce/checkout-js/pull/1086/files#diff-0ad410b99a894433eb62773917d0c31ce5e3878ffdb6f22bd228f1fad9793e0eR73 will be false.

Reply:
From what I understand we want to return the V1 component if id is ApplePay and gateway is Mollie? So in that case even if the component is resolved we should return the mollie component?

Reply:
well you're right, let me change it